src/gui/socketapi/socketapi.cpp

```diff
@@ -450,17 +450,30 @@ void SocketApi::slotUpdateFolderView(Folder *f)
 
     if (f) {
         // do only send UPDATE_VIEW for a couple of status
-        if (f->syncResult().status() == SyncResult::SyncPrepare
-            || f->syncResult().status() == SyncResult::Success
-            || f->syncResult().status() == SyncResult::Paused
-            || f->syncResult().status() == SyncResult::Problem
-            || f->syncResult().status() == SyncResult::Error
-            || f->syncResult().status() == SyncResult::SetupError) {
-            QString rootPath = removeTrailingSlash(f->path());
+        switch (f->syncResult().status()) {
+        case SyncResult::SyncPrepare:
+            Q_FALLTHROUGH();
+        case SyncResult::Success:
+            Q_FALLTHROUGH();
+        case SyncResult::Paused:
+            Q_FALLTHROUGH();
+        case SyncResult::Problem:
+            Q_FALLTHROUGH();
+        case SyncResult::Error:
+            Q_FALLTHROUGH();
+        case SyncResult::SetupError: {
+            const QString rootPath = removeTrailingSlash(f->path());
             broadcastStatusPushMessage(rootPath, f->syncEngine().syncFileStatusTracker().fileStatus(""));
 
-            broadcastMessage(buildMessage(QLatin1String("UPDATE_VIEW"), rootPath));
-        } else {
+            broadcastMessage(buildMessage(QStringLiteral("UPDATE_VIEW"), rootPath));
+        } break;
+        case OCC::SyncResult::Undefined:
+            Q_FALLTHROUGH();
+        case OCC::SyncResult::NotYetStarted:
+            Q_FALLTHROUGH();
+        case OCC::SyncResult::SyncRunning:
+            Q_FALLTHROUGH();
+        case OCC::SyncResult::SyncAbortRequested:
             qCDebug(lcSocketApi) << "Not sending UPDATE_VIEW for" << f->alias() << "because status() is" << f->syncResult().status();
         }
```

Comment:
if you checked all cases, maybe a `default: Q_UNREACHABE();` ?

Reply:
I omit the default on purpose to get compiler warnings

Reply:
we should rather make unhandled cases Werror